couchbase/cluster.hxx

```diff
@@ -99,29 +97,34 @@ class cluster
     template<typename Handler>
     void open_bucket(const std::string& bucket_name, Handler&& handler)
     {
-        if (buckets_.find(bucket_name) != buckets_.end()) {
+        auto b = find_bucket_by_name(bucket_name);
```

Comment:
We need to keep the lock between finding the bucket and adding it to the list.  Specifically the order that's probably ideal is to lock the lock, search for the bucket, if its not there, we should create a new bucket instance, then add it to the map, then unlock and finally bootstrap the connection. Otherwise there is a race where two of the same bucket could end up being bootstrapped.

Reply:
New version addresses this issue.